pyproject.toml

```diff
@@ -44,6 +44,7 @@ dev = [
 
     # code quality
     "black~=24.4.2",
+    "flake8~=7.1.0",
```

Comment:
why do we still use it along with `ruff`?

Reply:
Good point, removing flake8. This was leftover from earlier where we still had flake8 running in the commands, but had dropped the requirement